app/controllers/casebooks/books_controller.rb

```diff
@@ -0,0 +1,60 @@
+require 'net/http'
+require 'uri'
+
+class Casebooks::BooksController < ApplicationController
+  layout 'casebooks'
+  before_action :find_book, except: [:new, :create]
+  before_action :canonical_redirect, except: [:new, :create]
+  before_action :page_title
+
+  def new
+    @book = current_user.books.owned.unmodified.first ||
+      Casebook::Book.create(public: false, collaborators: [Casebook::Collaborator.new(user: current_user, role: 'owner')])
+    logger.debug @book.errors.inspect
+    redirect_to edit_book_path @book
+  end
+
+  def edit
+    @casebook = @book
+    render 'casebooks/edit'
+  end
+
+  def show
+    @casebook = @book
+    render 'casebooks/show'
+  end
+
+  def update
+    @book.update book_params
+    return redirect_to book_sections_path @book if @book.valid?
+    render 'casebooks/books/edit'
+  end
+
+  private
+
+  def book_params
+    params.require(:casebook_book).permit(:title, :subtitle, :ordinals, :headnote)
+  end
+
+  def find_book
+    @book = Casebook::Book.find params[:id]
+  end
+
+  def canonical_redirect
```

Comment:
The idea here is if we have a descriptive id in the url, like `/casebooks/100-third-restatement-torts` for a casebook titled "The Third Restatement of Torts", we want the slug portion to be optional— so `/casebooks/100` works, or `/casebooks/100-torts-draft`, since the title may change and we don't want to break links. But we want to canonicalize the current slug, so that new URLs use it instead of the one the user followed there.

So `canonical_redirect` checks if the user followed the *currently* canonical URL to get here, and redirects them if not. This is obviously only useful to do for GET routes, nothing that a user wouldn't be linking to.